codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/server/protocol/JsonDeserializerGenerator.java

```diff
@@ -243,20 +246,21 @@ yield goTemplate("""
                             ).compose(false)
                     ));
             case TIMESTAMP -> goTemplate("""
-                    dts, err := $T(serializedValue)
+                    dts, err := $T(av)
                     if err != nil {
-                        return nil, err
+                        return time.Time{}, err
                     }
                     return dts, nil
                     """, SmithyGoTypes.Time.ParseDateTime);
             default ->
-                    throw new CodegenException("Unsupported: " + shape.getType());
+                    throw new CodegenException("Unsupported: " + shape.getType() + " ShapeID: " + shape.getId());
         };
     }
 
-    private GoWriter.Writable generateDeserializeIntegral(String ident, String castTo, long min, long max) {
+    private GoWriter.Writable generateDeserializeIntegral(String ident, String castTo, String typecast,
```

Comment:
The way you've extended this doesn't seem right. This is called `generateDeserializeIntegral` and it's written to handle the integral types (int8,int16,etc) - but you've now overloaded it to do both integrals _and_ floating-point values. They should almost certainly be separate.